python/taichi/lang/__init__.py

```diff
@@ -552,6 +552,14 @@ def init(arch=None,
         _ti_core.set_tmp_dir(locale_encode(prepare_sandbox()))
     print(f'[Taichi] Starting on arch={_ti_core.arch_name(ti.cfg.arch)}')
 
+    # Torch based ndarray on opengl backend allocates memory on host instead of opengl backend.
+    # So it won't work.
+    if ti.cfg.arch == opengl and ti.cfg.ndarray_use_torch:
+        ti.warn(
```

Comment:
Just raise an error?

Reply:
@k-ye since `ndarray_use_torch=true` is default mode, making it an error might is a bit inconvenient for users. Esp when they don't know about `ndarray_use_torch`. So I might prefer keep this as a warning and turn it into an error when ndarray_use_torch=false becomes default?

Reply:
SG!